packages/amplify-util-uibuilder/src/commands/generateComponents.ts

```diff
@@ -34,10 +37,66 @@ export const run = async (context: $TSContext, eventType: 'PostPush' | 'PostPull
       studioClient.isGraphQLSupported ? getAmplifyDataSchema(context) : Promise.resolve(undefined),
     ]);
 
-    const nothingWouldAutogenerate =
-      !dataSchema || !studioClient.metadata.autoGenerateForms || !studioClient.isGraphQLSupported || !studioClient.isDataStoreEnabled;
+    let canCodegenGraphqlComponents = false;
+    let apiConfiguration: ApiConfiguration = {
+      dataStoreConfig: {},
+    };
```

Comment:
This generateComponents function could be a lot cleaner with a helper function to get the api configuration.